### DISCUSSION TITLE
What is the correct way to configure a static html page that contains js and css

### DISCUSSION THREAD

**Discussion by salmin89** [crxjs/chrome-extension-tools#730] [Q&A]
Hi!

I want to have a static page that can be opened through a link that links to: chrome-extension://{{extension}}/logViewerPage.html
I don't currently use a framework for this page. What is the correct way to setup CSS and JS for this static page?

What I currently have works, but I'm getting an error about the way I import the css file:
> Default and named imports from CSS files are deprecated. Use the ?inline query instead. For example: import css from './logViewerPage.css?inline'

I have tried adding ?inline but it doesn't work. Is there any example of how to properly do this?

-----

What I currently have:

**logViewerPage.html**
```
<script src="logViewerPage.js" type="module"></script>
```

**logViewerPage.js**
```
import css from './logViewerPage.css';
```

**vite.config.js**
```
export default defineConfig({
  plugins: [crx({ manifest })],
  build: {
    rollupOptions: {
      input: {
        logViewerPage: "logViewerPage.html"
      }
    }
  }
});
```

**manifest.json:**
```
 web_accessible_resources: [
      {
        matches: ['<all_urls>'],
        resources: ['src/html/logViewerPage.js', 'src/html/logViewerPage.css'],
        use_dynamic_url: true,
      },
    ],
```

**Comment by rileytaylor** [comment]
A few tips that might help. I've not tried this out without a framework but Vite should be handling things similarly for dev/build output regardless.

**manifest.json**

reference the html page in web_accessible_resources, not the js/css:

```json
"web_accessible_resources": [
  {
    "resources": [ "logViewerPage.html" ],
    "matches": [ "<all_urls>" ]
  }
]
```

Notice that the html file import in the manifest is not in a nested folder. If you use nested folders, you'll have the same nesting in the route in chrome... so `./html/logViewerPage.html` becomes `chrome-extension://{{...}}/html/logViewerPage.html`. There are ways around this [using a vite plugin to access some hooks provided by crxjs](https://github.com/crxjs/chrome-extension-tools/discussions/652). I haven't tested that yet, but it is possible.

**vite.config.js**

Make sure you have the right path to your html file. I like to use path.resolve() to be extra sure:

```js
import { resolve } from 'path' // depending on your ecmascript and node version the way you import may be different

build: {
  rollupOptions: {
    input: {
      logViewerPage: resolve(__dirname, 'logViewerPage.html')
    }
  }
}
```

**logViewerPage.html**

You're going to have a better time with a fully defined html file:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Log Viewer Page</title>
  </head>
  <body>
    <script type="module" src="./html/logViewerPage.js"></script>
  </body>
</html>
```

Ignore this suggestion if you just didn't copy in the entire thing.

**logViewerPage.js**

In vue it works a lot better to import the css this way. I've never seen an `?inline` recommendation like that outside of content scripts.

```js
import './logViewerPage.css';
```

**Comment by salmin89** [comment]
Thank you for this great answer!

I had to tweak it a little bit since my solution is structured like this:
```
project
│   vite.config.js
│   manifest.config.ts
│   tsconfig.json
│   package.json    
│
└───src
│   │   logViewerPage.html
│   │   ...
│   └─── ...
```

In vite.config.js I needed to specify the src-folder:
```
logViewerPage: resolve(__dirname, './src/logViewerPage.html')
```

not sure if this is because I need to specify src as rootDir in package.json or tsconfig?

either way, the provided example you gave me was very helpful :)

**Reply by rileytaylor** [reply]
Glad it got you there!
